### PR TITLE
fix(mock): use one-sided bounds when only minLength/maxLength or minItems/maxItems is specified

### LINKED ISSUE #3121: Incorrect string/array min value in MSW mocks when only maxLength/maxItems is specified
## Description

When generating MSW mocks, if a schema specifies only `maxLength` (without `minLength`) or only `maxItems` (without `minItems`), the generator incorrectly falls back to the global `stringMin`/`arrayMin` default for the unspecified bound.

With the default `stringMin: 10`, a schema with `maxLength: 5` produces:

```ts
faker.string.alpha({ length: { min: 10, max: 5 } })
```

`min > max` causes faker to throw at runtime. Even when `maxLength >= 10`, it is semantically wrong — the schema implies no minimum constraint, but the global default is applied anyway. The same issue occurs symmetrically: `minLength` without `maxLength` causes the global `stringMax` to be used as the upper bound, which may be less than `minLength`.

## Configuration (orval.config)

```ts
import { defineConfig } from 'orval';

export default defineConfig({
  api: {
    output: {
      target: './src/api/endpoints.ts',
      mock: { type: 'msw' },
    },
    input: { target: './openapi.yaml' },
  },
});
```

## Environment

Run the command above and paste the output here.

## Expected behavior

When only `maxLength: 5` is set, the generated mock should use only an upper bound:

```ts
faker.string.alpha({ length: { max: 5 } })
```

When only `minLength: 30` is set:

```ts
faker.string.alpha({ length: { min: 30 } })
```

Global defaults (`stringMin`, `stringMax`, `arrayMin`, `arrayMax`) should only be applied when **neither** bound is specified in the schema.

## Actual behavior

With `maxLength: 5` and the default `stringMin: 10`:

```ts
faker.string.alpha({ length: { min: 10, max: 5 } })
// ⚠️ faker throws — min > max
```

## OpenAPI document (minimal, if applicable)

```yaml
openapi: 3.0.3
info:
  title: Demo
  version: 0.0.0
paths:
  /items:
    get:
      operationId: getItems
      responses:
        '200':
          description: OK
          content:
            application/json:
              schema:
                type: object
                properties:
                  shortCode:
                    type: string
                    maxLength: 5
                  longDescription:
                    type: string
                    minLength: 30
                  tags:
                    type: array
                    maxItems: 2
                    items:
                      type: string
```

## Additional context

The bug is in `packages/mock/src/faker/getters/scalar.ts`. The fix is to conditionally apply global defaults only when neither bound is schema-specified:

```ts
// Before
const strMin = (item.minLength ?? safeMockOptions.stringMin) as number | undefined;
const strMax = (item.maxLength ?? safeMockOptions.stringMax) as number | undefined;

// After
const strMin = (item.minLength ?? (item.maxLength !== undefined ? undefined : safeMockOptions.stringMin)) as number | undefined;
const strMax = (item.maxLength ?? (item.minLength !== undefined ? undefined : safeMockOptions.stringMax)) as number | undefined;
```

Fix is available in PR #3120.

### PR DESCRIPTION
**Pull request by wadakatu**
Closes #3121
Supersedes #3120 (taking over per @melloware's request)

## Summary

When a schema specifies only `maxLength`/`maxItems` (without a corresponding min) or only `minLength`/`minItems` (without a corresponding max), the mock scalar generator was falling back to global defaults (`stringMin: 10`, `arrayMin: 1`, etc.) for the unspecified bound. This could produce invalid faker calls like `faker.string.alpha({length: {min: 10, max: 5}})` where `min > max`, causing faker to throw at runtime.

## What changed

- **Arrays** (`faker.number.int`): drop the unspecified bound entirely since faker accepts partial `{ min }` / `{ max }` options.
- **Strings** (`faker.string.alpha`): faker's `length: { min, max }` form requires BOTH bounds, so the global default fills the missing side only when it doesn't invert the range. When it would (e.g., `maxLength: 5` with default `stringMin: 10`), the explicit schema bound is reused on both sides — no values are invented that the user didn't supply.

## Notes on the takeover

This builds on the work in #3120 by @youngcm2 (credited via `Co-authored-by`). The previous iteration produced `{ max: 5 }`-only output that fails TypeScript checking under faker's strict `length` type for `faker.string.alpha`, which is why CI never went green. The clamp approach above keeps the original intent (no global default applied when the schema is one-sided) while satisfying faker's required `{ min, max }` pair for strings.

The CodeRabbit concern about `maxLength: 0 → {min: 1, max: 0}` is also addressed: with the clamp approach, `maxLength: 0` produces `{min: 0, max: 0}` (covered by a new unit test and a snapshot entry).

## Test plan

- [x] `bun run test` — all package unit tests pass (incl. 9 new scalar cases covering both variable and clamp paths for each side and the `maxLength: 0` edge case)
- [x] `bun run lint` — all packages clean
- [x] `bun run test:cli` — generated mock code typechecks for all 16 clients (previously failed silently due to the `length: { max }` shape)
- [x] `bun run --filter orval-tests test:snapshots` — all 4279 snapshots pass; `mock-constraints` extended to cover variable/clamp/zero paths

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved mock generation for strings and arrays with one-sided constraints; handles clamping, zero-max edge cases, and avoids emitting incomplete length bounds.

* **Tests**
  * Expanded coverage for constraint combinations, clamping behavior, and interactions with global mock defaults.

* **Chores**
  * Added a mock constraints fixture, config entry, and generated mock handlers/snapshots to support the new tests.

[Review Change Stack](https://app.coderabbit.ai/change-stack/orval-labs/orval/pull/3451?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: packages/mock/src/faker/getters/scalar.ts [hunk 1]
```diff
@@ -322,12 +322,14 @@ export function getMockScalar({
...
-        | number
-        | undefined;
+      const arrMin = (item.minItems ??
+        (item.maxItems === undefined
+          ? safeMockOptions.arrayMin
+          : undefined)) as number | undefined;
+      const arrMax = (item.maxItems ??
+        (item.minItems === undefined
+          ? safeMockOptions.arrayMax
+          : undefined)) as number | undefined;
```
[the remark this answers is not in the document]
Comment (wadakatu) [reply packages/mock/src/faker/getters/scalar.ts]: Good catch — addressed in 273198c8a. Mirrored the same clamp strategy from the string case so `minItems`-only no longer leaves `max` undefined (which could fall through to faker's internal default upper bound and produce very large arrays). When `safeMockOptions.arrayMax` is compatible we use it as the upper bound, otherwise we clamp to `minItems`. Added unit tests for the four paths and the snapshot is updated.

File: packages/mock/src/faker/getters/scalar.ts [hunk 2]
```diff
@@ -345,12 +347,38 @@ export function getMockScalar({
...
+      let strMax: number | undefined;
+      if (schemaMax !== undefined) {
+        strMax = schemaMax;
+      } else if (schemaMin === undefined) {
+        strMax = globalMax;
+      } else if (globalMax === undefined || globalMax < schemaMin) {
+        strMax = schemaMin;
+      } else {
+        strMax = globalMax;
+      }
```
[the remark this answers is not in the document]
Comment (wadakatu) [reply packages/mock/src/faker/getters/scalar.ts:405]: Addressed in 273198c8a. The emit step now only adds the `length: { min, max }` argument when both bounds could be resolved into a complete pair. When only one side is determinable (e.g., no schema bound and only one of `stringMin`/`stringMax` configured) we now fall back to `faker.string.alpha()` with no `length` argument, which avoids emitting a one-sided length object that would fail faker's required-pair type. Added a unit test covering the asymmetric-global case.

**Review by melloware — APPROVED** [review]
